src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ProcedureRequestMapper.java

```diff
@@ -1,12 +1,118 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Organization;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Practitioner;
 import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ReportStatusCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.PatientClinicalInfo;
+import uk.nhs.digital.nhsconnect.lab.results.utils.ResourceFullUrlGenerator;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.hl7.fhir.dstu3.model.ProcedureRequest.ProcedureRequestStatus;
+import org.hl7.fhir.dstu3.model.ProcedureRequest.ProcedureRequestIntent;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class ProcedureRequestMapper {
-    public ProcedureRequest mapToProcedureRequest(final Message message) {
-        return new ProcedureRequest();
+    private final UUIDGenerator uuidGenerator;
+    private final ResourceFullUrlGenerator fullUrlGenerator;
+    private static final Map<ReportStatusCode, ProcedureRequestStatus> STATUS_CODE_MAPPING = Map.of(
+        ReportStatusCode.UNSPECIFIED, ProcedureRequestStatus.UNKNOWN
+    );
+
+    public Optional<ProcedureRequest> mapToProcedureRequest(
+        final Message message,
+        Patient patient,
+        Practitioner requestingPractitioner,
+        Organization requestingOrganization,
+        Practitioner performingPractitioner,
+        Organization performingOrganization
```

Comment:
From what I can see, the members of these pairs are treated exactly the same, and if the organization is present the practitioner is ignored. Therefore the parameter could be at a higher level of abstraction (`DomainResource`) and the `PathologyRecordMapper` can use `optionalOrganization.or(() -> optionalPractitioner)` to derive it.

```
public Optional<ProcedureRequest> mapToProcedureRequest(final Message message, final Patient patient,
        final DomainResource requester, final DomainResource performer) {
    ...
}
```